pypesto/hierarchical/optimal_scaling/calculator.py

```diff
@@ -0,0 +1,186 @@
+"""Definition of an optimal scaling calculator class."""
+import copy
+from typing import Dict, List, Sequence, Tuple
+
+import numpy as np
+
+from ...C import FVAL, GRAD, HESS, MODE_RES, RDATAS, RES, SRES, X_INNER_OPT
+from ...objective.amici.amici_calculator import AmiciModel, AmiciSolver
+from ...objective.amici.amici_util import (
+    filter_return_dict,
+    init_return_values,
+)
+from .problem import OptimalScalingProblem
+from .solver import OptimalScalingInnerSolver
+
+try:
+    import amici
+    from amici.parameter_mapping import ParameterMapping
+except ImportError:
+    pass
+
+
+class OptimalScalingAmiciCalculator:
+    """A calculator is passed as `calculator` to the pypesto.AmiciObjective.
+
+    The object is called by the pypesto.AmiciObjective in ``call_unprocessed``
```

Comment:
```suggestion
    The object is called by :func:`pypesto.AmiciObjective.call_unprocessed`
```

yields proper links on readthedocs.